desci-server/src/controllers/doi/check.ts

```diff
@@ -0,0 +1,23 @@
+import { Request, Response } from 'express';
+
+import { BadRequestError, SuccessMessageResponse, SuccessResponse, doiService, logger } from '../../internal.js';
+
+export const checkMintability = async (req: Request, res: Response) => {
+  const { uuid } = req.params;
+  if (!uuid) throw new BadRequestError();
+  try {
+    await doiService.checkMintability(uuid);
+    new SuccessResponse(true).send(res);
+  } catch (err) {
+    logger.error(err, 'module:: checkMintability');
+    new SuccessResponse(false).send(res);
```

Comment:
Consider checking for expected errors and re-throwing the rest? Might be easier to discover issues if we don't swallow unexpected exceptions.

Reply:
I doesn't swallow unexpected expectations, the try catch is to reduce error thrown from the check into true of false because that's all the client needs. In another version of this, the errors will not be throw but returned as a checklist of what's needed to mint a Dpid and passed to the client.

Reply:
Yeah, that makes sense. I think what I'm missing is making it super visible to US that some other non-dpid-mintability exception occurred, so I should just integrate Sentry here to toss exceptions like that into.

Reply:
Yeah we can add sentry for logging but in this case we expect errors to be thrown here so that might not be a good case for sentry. Returning the possible errors as a checklist and Removing the try/catch will make total sense here so the error can be caught in the error handler middleware and reported via sentry.

Reply:
I'm going to report other exceptions not instanceof DoiError type with sentry